sklearnex/_device_offload.py

```diff
@@ -27,49 +27,51 @@
 from ._config import get_config
 
 
-def _get_backend(obj, queue, method_name, *data):
-    cpu_device = queue is None or queue.sycl_device.is_cpu
-    gpu_device = queue is not None and queue.sycl_device.is_gpu
+def _get_backend(obj, method_name, *data):
+    queue = SyclQueueManager.get_global_queue()
+    cpu_device = queue is None or getattr(queue.sycl_device, "is_cpu", True)
+    gpu_device = queue is not None and getattr(queue.sycl_device, "is_gpu", False)
 
     if cpu_device:
         patching_status = obj._onedal_cpu_supported(method_name, *data)
         if patching_status.get_status():
-            return "onedal", queue, patching_status
+            return "onedal", patching_status
         else:
-            return "sklearn", None, patching_status
+            return "sklearn", patching_status
 
     allow_fallback_to_host = get_config()["allow_fallback_to_host"]
 
     if gpu_device:
         patching_status = obj._onedal_gpu_supported(method_name, *data)
         if patching_status.get_status():
-            return "onedal", queue, patching_status
+            return "onedal", patching_status
         else:
+            SyclQueueManager.remove_global_queue()
             if allow_fallback_to_host:
                 patching_status = obj._onedal_cpu_supported(method_name, *data)
                 if patching_status.get_status():
-                    return "onedal", None, patching_status
+                    return "onedal", patching_status
                 else:
-                    return "sklearn", None, patching_status
+                    return "sklearn", patching_status
             else:
-                return "sklearn", None, patching_status
+                return "sklearn", patching_status
 
     raise RuntimeError("Device support is not implemented")
 
 
 def dispatch(obj, method_name, branches, *args, **kwargs):
-    q = _get_global_queue()
-    has_usm_data_for_args, q, hostargs = _transfer_to_host(q, *args)
-    has_usm_data_for_kwargs, q, hostvalues = _transfer_to_host(q, *kwargs.values())
+    queue = getattr(SyclQueueManager.get_global_queue(), "implementation", None)
```

Comment:
Assuming this PR passes initial review, I would like to improve `SyclQueueManager`. Right now it only acts as a global namespace, and the provided utilities (like `get_global_queue()`) don't have a perfect interface yet. The fact we need to query `getattr()` isn't nice. It's easy enough to refactor and pull this straight. But I'll only do that if my suggestions are generally accepted.